packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -63,14 +64,28 @@ export function getSPOAndGraphRequestIdsFromResponse(headers: { get: (id: string
     return additionalProps;
 }
 
+export interface IFacetCodes {
+    facetCodes?: string[];
+ }
+
+ function parseFacetCodes(response): string[] {
+    const stack: string[] = [];
+    let error = response.error;
+    while (error !== undefined) {
+        stack.push(error.code);
+        error = error.innerError;
+   }
+   return stack;
+}
+
 export function createOdspNetworkError(
     errorMessage: string,
     statusCode: number,
     retryAfterSeconds?: number,
     response?: Response,
     responseText?: string,
 ): OdspError {
```

Comment:
Can you please change return type as well to be OdspError & LoggingError & IFacetCodes ?